docs/ref/contrib/postgres/constraints.txt

```diff
@@ -0,0 +1,65 @@
+=======================================
+PostgreSQL specific exclude constraints
+=======================================
+
+.. module:: django.contrib.postgres.constraints
+   :synopsis: PostgreSQL specific database constraint
+
+The exclude constraints are like :class:`~django.db.models.CheckConstraint`,
+which is useful to ensure different integrity constraints on :ref:`range fields
+<range-fields>`, or tables that have implemented a range with two fields. They
+are declared in :attr:`Meta.constraints
+<django.db.models.Options.constraints>`, and detected automatically when you
+run :djadmin:`makemigrations`. The exclude constraint may be considered as an
+advanced unique constraint.
+
+You need to activate the `btree_gist extension
+<https://www.postgresql.org/docs/current/static/btree-gist.html>`_ on
+PostgreSQL before this constraint can be used as described in the
+documentation_. You can install it using the
+:class:`~django.contrib.postgres.operations.BtreeGistExtension` migration
+operation.
+
+The classic example discussed in the documentation_ uses a hotel and excludes
+overlapping reservations with check constraints. Additional constraints can be
+specified via :class:`~django.db.models.CheckConstraint`'s ``check`` argument.
+
+If the table you'd like to add an exclusion constraint has implemented the
+range with two fields, you may use a range
+:class:`~psycopg2:psycopg2.extras.Range` as the basis for a function::
+
+    .. code-block::
+
+        from django.contrib.postgres.fields import DateTimeRangeField
+        from django.db.models import Func
+
+
+        class TsTzRange(Func):
+            function = 'tstzrange'
+            output_field = DateTimeRangeField()
+            arity = 3
+
+The above example creates a function that you can use when creating the
+exclusion constraint.
+
+.. _documentation: https://www.postgresql.org/docs/current/static/sql-createtable.html#SQL-CREATETABLE-EXCLUDE
+
+``ExcludeConstraint``
+=====================
+
+.. class:: ExcludeConstraint(*, expressions=(), operators=(), index_type='GIST', check=None, name='')
```

Comment:
`check` -> `condition`?

Reply:
I replaced it consequently throughout the patch :+1: